src/scores/categorical/multicategorical_impl.py

```diff
@@ -121,10 +121,10 @@ def firm(  # pylint: disable=too-many-arguments
     reduce_dims = gather_dimensions(
         fcst.dims, obs.dims, reduce_dims=reduce_dims, preserve_dims=preserve_dims
     )  # type: ignore[assignment]
```

Comment:
Why is this type ignore more prescriptive?

Reply:
Earlier in the work I was putting more effort into trying to be as prescriptive as possible, but later I realised how many issues there were going to be, and how much the tools were going to struggle, so I went to less-prescriptive overrides in order to get through it. Being more prescriptive didn't seem to help mypy with understanding the code, so it's more about the user feedback than optimising the tool functionality.